src/test/unit-converter.test.ts

```diff
@@ -0,0 +1,246 @@
+      chai.expect(numberToString("-50")).to.equal("-50");
+      chai.expect(numberToString("-3.14")).to.equal("-3.14");
+      chai.expect(numberToString("0")).to.equal("0");
+    });
+
+    it("Should throw an error if string is invalid", async function () {
+      chai
+        .expect(() => numberToString("/42"))
+        .to.throw(Error)
+        .with.property(
+          "message",
+          "while converting number to string, invalid number value '/42', should be a number matching (^-?[0-9.]+)."
+        );
+      chai
+        .expect(() => numberToString("&42"))
+        .to.throw(Error)
+        .with.property(
+          "message",
+          "while converting number to string, invalid number value '&42', should be a number matching (^-?[0-9.]+)."
+        );
+      chai
+        .expect(() => numberToString(" 42"))
+        .to.throw(Error)
+        .with.property(
+          "message",
+          "while converting number to string, invalid number value ' 42', should be a number matching (^-?[0-9.]+)."
+        );
+    });
+
+    it("Should convert a number to string", async function () {
+      chai.expect(numberToString(42)).to.equal("42");
+      chai.expect(numberToString(8.15)).to.equal("8.15");
+      chai.expect(numberToString(-50)).to.equal("-50");
+      chai.expect(numberToString(-3.14)).to.equal("-3.14");
+      chai.expect(numberToString(0)).to.equal("0");
+    });
+
+    it("Should convert an object with toString method to string", async () => {
+      let customValue = "42";
+      const objWithToString = {
+        toString: () => customValue.toString(),
+        toTwos: () => "",
+        dividedToIntegerBy: () => "",
+      };
+      chai.expect(numberToString(objWithToString)).to.equal("42");
+      customValue = "8.15";
+      chai.expect(numberToString(objWithToString)).to.equal("8.15");
+      customValue = "-50";
+      chai.expect(numberToString(objWithToString)).to.equal("-50");
+      customValue = "-3.14";
+      chai.expect(numberToString(objWithToString)).to.equal("-3.14");
+      customValue = "0";
+      chai.expect(numberToString(objWithToString)).to.equal("0");
+    });
+
+    it("Should throw an error for invalid inputs", async () => {
+      chai
+        .expect(() => numberToString("invalid"))
+        .to.throw(Error)
+        .with.property(
+          "message",
+          "while converting number to string, invalid number value 'invalid', should be a number matching (^-?[0-9.]+)."
+        );
+      chai
+        .expect(() => numberToString({}))
+        .to.throw(Error)
+        .with.property(
+          "message",
+          "while converting number to string, invalid number value '[object Object]' type object."
+        );
+      chai
+        .expect(() => numberToString([]))
+        .to.throw(Error)
+        .with.property("message", "while converting number to string, invalid number value '' type object.");
+    });
+  });
+
+  describe("scaleDecimals", async function () {
```

Comment:
@juniormp The following case seems to be not covered here:

https://github.com/grindery-io/grindery-nexus-connector-web3/blob/907e72a3615ddf6326b63635199d835a2dc43c89/src/web3/evm/unitConverter.ts#L38

Can you add it?